src/cogent3/app/io_new.py

```diff
@@ -45,6 +49,62 @@
 __status__ = "Alpha"
 
 
+@define_app
+def pickle_it(data: SerialisableType) -> bytes:
+    """attempts to call a to_rich_dict() method, fallback to
+    pickling entire object"""
+    with contextlib.suppress(AttributeError):
+        data = data.to_rich_dict()
+
+    return pickle.dumps(data)
+
+
+@define_app
+def unpickle_it(data: bytes) -> SerialisableType:
+    return pickle.loads(data)
+
+
+@define_app
+class compressed:
+    def __init__(self, compressor: callable = gzip_compress):
+        """
+        Parameters
+        ----------
+        compressor
+            function for compressing bytes data, defaults to gzip
+        """
+        self.compressor = compressor
+
+    def main(self, data: bytes) -> bytes:
+        return self.compressor(data)
+
+
+@define_app
+class decompressed:
+    def __init__(self, decompressor: callable = gzip_decompress):
+        """
+        Parameters
+        ----------
+        decompressor
+            a function for decompression, defaults to the gzip decompress
+            function
+        """
+        self.decompressor = decompressor
+
+    def main(self, data: bytes) -> bytes:
+        return self.decompressor(data)
+
+
+@define_app
+class deserialised:
```

Comment:
Do we need a `serialiser` app?

Reply:
Perhaps `deserialised` is a bad name, since `pickle.loads()` is a deserialiser. This is specifically about "inflating" cogent3 objects. Will think on it.

Reply:
Addressed in new revisions